### PR TITLE
[ssip]: Update config generator

### PR DESCRIPTION
**Pull request by nazariig**
**HLD:** https://github.com/sonic-net/SONiC/pull/1002

**Example:**
```
root@sonic:/home/admin# cat /etc/rsyslog.conf | grep "^*.*"
*.* action(type="omfwd" target="10.0.0.5" protocol="udp" address="1.1.1.1" port="514")
*.* action(type="omfwd" target="10.0.0.6" protocol="udp" address="1.1.1.1" port="514" device="mgmt")
```

[sonic-utilities](https://github.com/Azure/sonic-utilities)
```
ea11b22 [sonic-bootchart] add sonic-bootchart (#2195)
8e5d478 [ssip]: Add CLI (#2191)
1dacb7f Replace pyswsssdk with swsscommon (#2251)
```

#### Why I did it
* To implement Syslog Source IP feature
* In order to include the following commit: `8e5d478 [ssip]: Add CLI (#2191)`

#### How I did it
* Updated syslog config template
* Advanced submodule

#### How to verify it
1. make configure PLATFORM=mellanox
2. make target/sonic-mellanox.bin

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111

#### Description for the changelog

* N/A

#### Link to config_db schema for YANG module changes

* [sonic-syslog.yang](https://github.com/Azure/sonic-buildimage/blob/master/src/sonic-yang-models/yang-models/sonic-syslog.yang)

#### A picture of a cute animal (not mandatory but encouraged)
```
      .---.        .-----------
     /     \  __  /    ------
    / /     \(  )/    -----
   //////   ' \/ `   ---
  //// / // :    : ---
 // /   /  /`    '--
//          //..\\
       ====UU====UU====
           '//||\\`
             ''``
```

### CODE REVIEW

File: files/image_config/rsyslog/rsyslog.conf.j2 [hunk 1]
```diff
@@ -72,3 +68,26 @@ $RepeatedMsgReduction on
...
+# Remote syslog logging
+#
+
+# The omfwd plug-in provides the core functionality of traditional message forwarding via UDP and plain TCP.
+# It is a built-in module that does not need to be loaded.
+
+{% if SYSLOG_SERVER is defined %}
+{% for server, data in SYSLOG_SERVER.items() %}
+{% set params_list = [] %}
+{% if 'source' in data %}
```
Comment (qiluo-msft) [inline files/image_config/rsyslog/rsyslog.conf.j2:82]: >source [](http://example.com/codeflow?start=7&length=6)

Please add unit test.

File: files/image_config/rsyslog/rsyslog.conf.j2 [hunk 2]
```diff
@@ -72,3 +68,26 @@ $RepeatedMsgReduction on
...
 ###############
+
+#
+# Remote syslog logging
+#
+
+# The omfwd plug-in provides the core functionality of traditional message forwarding via UDP and plain TCP.
+# It is a built-in module that does not need to be loaded.
+
+{% if SYSLOG_SERVER is defined %}
```
Comment (wen587) [inline files/image_config/rsyslog/rsyslog.conf.j2:79]: Could you add what the config looks like in the description?
For example, the current `rsyslog.conf` looks like the below:
```
admin@vlab-01:~$ cat /etc/rsyslog.conf  | grep "^*.*"
*.* @[10.0.0.5]:514;SONiCFileFormat
*.* @[10.0.0.6]:514;SONiCFileFormat

```

File: files/image_config/rsyslog/rsyslog.conf.j2 (hunk 1 above)
Comment (nazariig) [reply files/image_config/rsyslog/rsyslog.conf.j2:82]: @qiluo-msft can you please clarify which UTs are you referring to? Do we have a Framework?

File: files/image_config/rsyslog/rsyslog.conf.j2 (hunk 2 above)
Comment (nazariig) [reply files/image_config/rsyslog/rsyslog.conf.j2:79]: @wen587 done

File: files/image_config/rsyslog/rsyslog.conf.j2 (hunk 1 above)
Comment (qiluo-msft) [reply files/image_config/rsyslog/rsyslog.conf.j2:82]: Either sonic-config-engine unit test or sonic-mgmt test case.

**Comment by wen587** [comment]
Hi, I think this change will break `show runningconfiguration syslog`
https://github.com/Azure/sonic-utilities/blob/master/show/main.py#L1547-L1578
May need update in sonic-utilities.

File: files/image_config/rsyslog/rsyslog.conf.j2 (hunk 1 above)
Comment (nazariig) [reply files/image_config/rsyslog/rsyslog.conf.j2:81]: @qiluo-msft can we converge on this: https://github.com/Azure/sonic-mgmt/pull/5904 ?

Comment (nazariig) [reply files/image_config/rsyslog/rsyslog.conf.j2:81]: @qiluo-msft just a kind reminder

**Review by qiluo-msft — APPROVED** [review]

**Comment by nazariig** [comment]
The PR has a circular dependency on a `sonic-utilities` submodule update, hence need to be merged first

**Comment by liat-grozovik** [comment]
@nazariig why the submodule update is part of code changes on buildimage?
the description is not aligned same goes for the labels.
please clarify what exactly is expected to be included in this PR

**Comment by nazariig** [comment]
> @nazariig why the submodule update is part of code changes on buildimage? the description is not aligned same goes for the labels. please clarify what exactly is expected to be included in this PR

@qiluo-msft asked to incorporate submodule update into this PR to avoid circular dependency

**Review by qiluo-msft — APPROVED** [review]

**Comment by nazariig** [comment]
Contains changes from: https://github.com/Azure/sonic-buildimage/pull/11468

**Pull request merged**